src/modules/ecdsa_adaptor/dleq_impl.h

```diff
@@ -0,0 +1,172 @@
+#ifndef _SECP256K1_DLEQ_IMPL_H_
+#define _SECP256K1_DLEQ_IMPL_H_
+
+/* Remove terminating NUL bytes */
+static int algo16_len(const unsigned char *algo16) {
+    int algo16_len = 16;
+
+    /* Remove terminating null bytes */
+    while (algo16_len > 0 && !algo16[algo16_len - 1]) {
+        algo16_len--;
+    }
+    return algo16_len;
+}
+
+/* Modified bip340 nonce function */
+static int nonce_function_dleq(unsigned char *nonce32, const unsigned char *msg32, const unsigned char *key32, const unsigned char *algo16) {
+    secp256k1_sha256 sha;
+
+    if (algo16 == NULL) {
+        return 0;
+    }
+    secp256k1_sha256_initialize_tagged(&sha, algo16, algo16_len(algo16));
+    secp256k1_sha256_write(&sha, key32, 32);
+    secp256k1_sha256_write(&sha, msg32, 32);
+    secp256k1_sha256_finalize(&sha, nonce32);
+    return 1;
+}
+
+static void secp256k1_dleq_serialize_point(unsigned char *buf33, const secp256k1_ge *p) {
+    secp256k1_fe x = p->x;
+    secp256k1_fe y = p->y;
+
+    secp256k1_fe_normalize(&y);
+    buf33[0] = secp256k1_fe_is_odd(&y);
+    secp256k1_fe_normalize(&x);
+    secp256k1_fe_get_b32(&buf33[1], &x);
+}
+
+static int secp256k1_dleq_deserialize_point(secp256k1_ge *p, const unsigned char *buf33) {
+    secp256k1_fe x;
+
+    if (!secp256k1_fe_set_b32(&x, &buf33[1])) {
+        return 0;
+    }
+    if (buf33[0] > 1) {
+        return 0;
+    }
+    secp256k1_ge_set_xo_var(p, &x, buf33[0]);
```

Comment:
`secp256k1_ge_set_xo_var` can fail https://github.com/bitcoin-core/secp256k1/blob/6034a04fb1afe7d78dd367ec719d3ced9db2b05e/src/group_impl.h#L234 I think you should check the return.

Reply:
Absolutely, great catch. Pushed fix. There could be more issues like that because this PR doesn't include a complete test suite yet (as mentioned in the TODO list of `secp256k1_ecdsa_adaptor.h`.